src/libs/actions/PersonalDetails/index.js

```diff
@@ -235,6 +236,54 @@ function setPersonalDetails(details) {
     mergeLocalPersonalDetails(details);
 }
 
+/**
+ * Fetches the Currency preferences based on location
+ * @param {bool} withLocation
+ * @param {Object} details
+ */
+function fetchCurrencyPreferences(withLocation) {
+    let coords = {};
+    let currency = '';
+    if (withLocation) {
+        Geolocation.getCurrentPosition((position) => {
+            coords = {
+                longitude: position.coords.longitude,
+                latitude: position.coords.latitude,
+            };
+        });
+    }
+
+    API.GetPreferredCurrency({...coords})
+        .then((data) => {
+            currency = data.currency;
+        })
+        .then(() => API.GetCurrencyList())
+        .then((data) => {
+            const currencyList = JSON.parse(data.currencyList);
+            Onyx.merge(ONYXKEYS.CURRENCY_LIST, currencyList);
+            return currencyList;
+        })
+        .then((currencyList) => {
+            Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS,
+                {preferredCurrencyCode: currency, preferredCurrencySymbol: currencyList[currency].symbol});
+        })
+        .catch(error => console.debug(error));
+}
+
+/**
+ * Gets the currency list from the network
+ *
+ * @param {Object} details
```

Comment:
Can you add the return doc

Reply:
Changed the comment to reflect it's real function.